src/test/java/uk/gov/justice/probation/courtcaseservice/service/OffenderServiceTest.java

```diff
@@ -83,6 +92,7 @@ void beforeEach() {
     void whenGetOffender_returnOffenderWithConvictionsDocumentsNotFiltered() {
         when(offenderRestClient.getProbationRecordByCrn(CRN)).thenReturn(Mono.just(ProbationRecord.builder().crn(CRN).build()));
         when(offenderRestClient.getConvictionsByCrn(CRN)).thenReturn(Mono.just(singletonList(conviction)));
+        when(offenderRestClient.getBreaches(CRN, CONVICTION_ID)).thenReturn(Mono.just(singletonList(breach)));
```

Comment:
Would like to see the Mockito verify call to prove that this happened. Mockito runs in non-strict mode and it's good to know that each and every "when" is necessary and doing as expected.

Reply:
can do, although mockito does enforce this anyway. if you add a stub for something that isn't called the test fails with an exception like this:

```
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at uk.gov.justice.probation.courtcaseservice.service.OffenderServiceTest.givenOffenderNotFound_whenGetOffender_thenThrowException(OffenderServiceTest.java:144)
```